rewrite-maven/src/main/java/org/openrewrite/maven/tree/ResolvedPom.java

```diff
@@ -978,10 +979,20 @@ public List<ResolvedDependency> resolveDependencies(Scope scope, Map<GroupArtifa
         for (Dependency requestedDependency : getRequestedDependencies()) {
             Dependency d = getValues(requestedDependency, 0);
             Scope dScope = Scope.fromName(d.getScope());
-            if (dScope == scope || dScope.transitiveOf(scope) == scope) {
-                // TODO can we always use the Map.put approach? Using the latest one is Maven specific, but this resolving is also used for gradle which does use highest version.
-                //  We could introduce a ResolutionStrategy to handle this and use Map.merge where we take later occurring one for LAST_WINS/MAVEN and higher version one for LATEST_WINS/GRADLE
-                rootDependencies.put(d.getGav().asGroupArtifact(), new DependencyAndDependent(requestedDependency, Scope.Compile, null, requestedDependency, this));
+            rootDependencies.remove(d.getGav().asGroupArtifact());
+            // TODO can we always use the Map.put approach? Using the latest one is Maven specific, but this resolving is also used for gradle which does use highest version.
+            //  We could introduce a ResolutionStrategy to handle this and use Map.merge where we take later occurring one for LAST_WINS/MAVEN and higher version one for LATEST_WINS/GRADLE
+            switch (scope) {
+                case Compile:
+                    maybePutRootDependency(scope, requestedDependency, dScope, rootDependencies, d, s -> s == Scope.Provided);
+                    break;
+                case Runtime:
+                case Provided:
+                    maybePutRootDependency(scope, requestedDependency, dScope, rootDependencies, d);
```

Comment:
Nitpick, but I think I would just do:


```suggestion
                    maybePutRootDependency(scope, requestedDependency, dScope, rootDependencies, d, null);
```